src/Futhark/Compiler/Program.hs

```diff
@@ -51,8 +51,8 @@ import Language.Futhark.Prop (isBuiltin)
 import Language.Futhark.Semantic
 import qualified Language.Futhark.TypeChecker as E
 import Language.Futhark.Warnings
-import Language.LSP.Types (normalizedFilePathToUri, toNormalizedFilePath)
-import Language.LSP.VFS (VFS (vfsMap), virtualFileText)
+import Language.LSP.Types (NormalizedUri, normalizedFilePathToUri, toNormalizedFilePath)
+import Language.LSP.VFS (VirtualFile, virtualFileText)
```

Comment:
I don't think this is appropriate.  Something is wrong if this module has to depend on LSP types.  You should define the VFS contents as `M.Map FilePath T.Text` (with the `FilePath` understood as an absolute path).  The LSP server itself then has to preprocess the `VirtualFile`s and URIs and whatever *before* calling anything in this module.  It is important to maintain layering, and putting LSP details inside this foundational module is a layering violation.

Reply:
Great point, removed all imports from lsp package in `Program.hs`